src/pkcs11/mechanism.c

```diff
@@ -1134,6 +1259,151 @@ sc_pkcs11_deri(struct sc_pkcs11_session *session,
 }
 
 
+/*
+ * Initialize a encrypt operation
+ */
+static CK_RV
+sc_pkcs11_encrypt_init(sc_pkcs11_operation_t *operation,
+			struct sc_pkcs11_object *key)
+{
+	struct signature_data *data;
```

Comment:
The usage of `signature_data` is confusing here. If I see right, only the `key` part is used. Would it make sense to assign the key directly into the `operation->priv_data`? Or create generic `operation_data` if we envision there will be some more data needed to share between the functions?

Reply:
We already have same confusing name in `sc_pkcs11_decrypt_init()`. It doesn't make sense to fix this in this PR just because of the new operation `sc_pkcs11_encrypt_init ()`, we have to fix it globally (so also for the function `sc_pkcs11_decrypt_init ()`) 

`````
static CK_RV
sc_pkcs11_decrypt_init(sc_pkcs11_operation_t *operation,
                        struct sc_pkcs11_object *key)
{
        struct signature_data *data;
        CK_RV rv;

        if (!(data = new_signature_data()))
                return CKR_HOST_MEMORY;
`````
> Would it make sense to assign the key directly into the operation->priv_data?

This solution complicates the freeing of memory (in `sc_pkcs11_new_fw_mechanism ()` we use `mt-> release = sc_pkcs11_signature_release;`)

It might be a good idea to rename `struct signature_data` to` operation_data` and similarly rename functions that operate with it: `new_signature_data()`, `signature_data_release()`, `signature_data_buffer_append()` and of course `sc_pkcs11_signature_release()`.  

The question is whether this renaming will not impair the readability of the code in sign and verify operations. I believe that this renaming should be done outside this PR.

Reply:
could you create a PR with the suggested fix to the naming?